.github/workflows/ci.yaml

```diff
@@ -42,3 +43,24 @@ jobs:
         with:
           name: test-results
           path: ./project-tests.log
+  build-and-push:
+    needs: [project-tests]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to GHCR
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I haven't added a secret anywhere yet. Just replicating what we have here: https://github.com/elifesciences/enhanced-preprints-client/blob/master/.github/workflows/ci.yaml#L48-L63